python/ray/llm/_internal/serve/configs/server_models.py

```diff
@@ -339,17 +339,28 @@ def get_engine_config(self):
 
         LLMConfig not only has engine config but also deployment config, etc.
         """
+        # Note (genesu): This is important that we cache the engine config as the
+        # `hf_model_id` attribute on the engine config will be set based on whether
+        #  the model is downloaded from a remote storage and will be set to the
+        #  local path of the model. This is important for vLLM not going to Hugging
+        #  Face to download the model again after it's already downloaded during node
+        #  initialization step.
+        if hasattr(self, "_engine_config") and self._engine_config:
+            return self._engine_config
+
         if self.llm_engine == LLMEngine.vLLM:
             from ray.llm._internal.serve.deployments.llm.vllm.vllm_models import (
                 VLLMEngineConfig,
             )
 
-            return VLLMEngineConfig.from_llm_config(self)
+            self._engine_config = VLLMEngineConfig.from_llm_config(self)
```

Comment:
Why it's not sufficient if we make sure we don't download the model files again if the local one exists? Intuitively I feel this caching should be handled in `from_llm_config` instead of here, but I'm likely missing something.